src/test/java/org/folio/fqm/service/PermissionsRegularServiceTest.java

```diff
@@ -7,40 +7,43 @@
 import static org.mockito.Mockito.when;
 
 import java.util.*;
-import org.apache.commons.lang3.NotImplementedException;
 import org.folio.fqm.client.ModPermissionsClient;
+import org.folio.fqm.client.ModRolesKeyClockClient;
 import org.folio.fqm.exception.MissingPermissionsException;
 import org.folio.querytool.domain.dto.EntityType;
 import org.folio.querytool.domain.dto.EntityTypeSource;
 import org.folio.spring.FolioExecutionContext;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 class PermissionsRegularServiceTest {
 
   private final FolioExecutionContext context = mock(FolioExecutionContext.class);
   private final ModPermissionsClient modPermissionsClient = mock(ModPermissionsClient.class);
+  private final ModRolesKeyClockClient modRolesKeyclockClient = mock(ModRolesKeyClockClient.class);
   private final EntityTypeFlatteningService entityTypeFlatteningService = mock(EntityTypeFlatteningService.class);
   private final PermissionsRegularService permissionsService = new PermissionsRegularService(
     context,
     modPermissionsClient,
+    modRolesKeyclockClient,
     entityTypeFlatteningService
   );
   private static final String TENANT_ID = "tenant_01";
 
-  @BeforeEach
-  void setUp() {
-    permissionsService.isEureka = false; // Force the service to use the mod-permissions client
-  }
-
   void setUpMocks(String... permissions) {
     var userId = UUID.randomUUID();
     when(context.getUserId()).thenReturn(userId);
     when(context.getTenantId()).thenReturn(TENANT_ID);
-    when(modPermissionsClient.getPermissionsForUser(TENANT_ID, userId.toString()))
-      .thenReturn(new ModPermissionsClient.UserPermissions(List.of(permissions), permissions.length));
+
+    if (permissionsService.isEureka) {
```

Comment:
it feels like we should have tests for both cases...

Reply:
then it would go in the else block right or you mean something else

Reply:
i mean we'd need one test with `isEureka = true` and another `isEureka = false`

Reply:
I hope the update makes you happy @ncovercash :)